src/plugins/procmon/ntstatus.cpp

```diff
@@ -0,0 +1,5551 @@
+        [FACILITY_TERMINAL_SERVER]     = "TERMINAL_SERVER",
+        [FACILITY_USB_ERROR_CODE]      = "USB_ERROR_CODE",
+        [FACILITY_HID_ERROR_CODE]      = "HID_ERROR_CODE",
+        [FACILITY_FIREWIRE_ERROR_CODE] = "FIREWIRE_ERROR_CODE",
+        [FACILITY_CLUSTER_ERROR_CODE]  = "CLUSTER_ERROR_CODE",
+        [FACILITY_ACPI_ERROR_CODE]     = "ACPI_ERROR_CODE",
+        [FACILITY_SXS_ERROR_CODE]      = "SXS_ERROR_CODE",
+        [FACILITY_TRANSACTION]         = "TRANSACTION",
+        [FACILITY_COMMONLOG]           = "COMMONLOG",
+        [FACILITY_VIDEO]               = "VIDEO",
+        [FACILITY_FILTER_MANAGER]      = "FILTER_MANAGER",
+        [FACILITY_MONITOR]             = "MONITOR",
+        [FACILITY_GRAPHICS_KERNEL]     = "GRAPHICS_KERNEL",
+        [FACILITY_DRIVER_FRAMEWORK]    = "DRIVER_FRAMEWORK",
+        [FACILITY_FVE_ERROR_CODE]      = "FVE_ERROR_CODE",
+        [FACILITY_FWP_ERROR_CODE]      = "FWP_ERROR_CODE",
+        [FACILITY_NDIS_ERROR_CODE]     = "NDIS_ERROR_CODE",
+        [FACILITY_TPM]                 = "TPM",
+        [FACILITY_RTPM]                = "RTPM",
+        [FACILITY_HYPERVISOR]          = "HYPERVISOR",
+        [FACILITY_IPSEC]               = "IPSEC",
+        [FACILITY_VIRTUALIZATION]      = "VIRTUALIZATION",
+        [FACILITY_VOLMGR]              = "VOLMGR",
+        [FACILITY_BCD_ERROR_CODE]      = "BCD_ERROR_CODE",
+        [FACILITY_WIN32K_NTUSER]       = "WIN32K_NTUSER",
+        [FACILITY_WIN32K_NTGDI]        = "WIN32K_NTGDI",
+        [FACILITY_RESUME_KEY_FILTER]   = "RESUME_KEY_FILTER",
+        [FACILITY_RDBSS]               = "RDBSS",
+        [FACILITY_BTH_ATT]             = "BTH_ATT",
+        [FACILITY_SECUREBOOT]          = "SECUREBOOT",
+        [FACILITY_AUDIO_KERNEL]        = "AUDIO_KERNEL",
+        [FACILITY_VSM]                 = "VSM",
+        [FACILITY_VOLSNAP]             = "VOLSNAP",
+        [FACILITY_SDBUS]               = "SDBUS",
+        [FACILITY_SHARED_VHDX]         = "SHARED_VHDX",
+        [FACILITY_SMB]                 = "SMB",
+        [FACILITY_INTERIX]             = "INTERIX",
+        [FACILITY_SPACES]              = "SPACES",
+        [FACILITY_SECURITY_CORE]       = "SECURITY_CORE",
+        [FACILITY_SYSTEM_INTEGRITY]    = "SYSTEM_INTEGRITY",
+        [FACILITY_LICENSING]           = "LICENSING",
+        [FACILITY_PLATFORM_MANIFEST]   = "PLATFORM_MANIFEST",
+        [FACILITY_APP_EXEC]            = "APP_EXEC"
+    };
+    unsigned severity = (status >> 30) & 3;
+    unsigned customer = (status >> 29) & 1;
+    unsigned facility = (status >> 16) & 0x00000FFFL;
+    unsigned code     = (status & 0x0000FFFFL);
+
+    snprintf(buffer, buf_size, "%s:%d:%s:0x%x",
+             severities[severity],
+             customer,
+             facility < FACILITY_MAXIMUM_VALUE ? facilities[facility] : "UNKNOWN",
+             code);
+
+    return buffer;
+}
+
+const char* ntstatus_to_string(ntstatus_t status)
+{
+    switch (status)
```

Comment:
Instead of a switch-return this could be easily turned into an array of strings.